src/inference/src/check_network_batchable.cpp

```diff
@@ -0,0 +1,79 @@
+#include "check_network_batchable.hpp"
+
+#include "dimension_tracker.hpp"
+#include "ie_ngraph_utils.hpp"
+#include "ngraph/opsets/opset.hpp"
+#include "openvino/op/detection_output.hpp"
+#include "openvino/op/ops.hpp"
+#include "openvino/pass/manager.hpp"
+#include "transformations/common_optimizations/dimension_tracking.hpp"
+#include "transformations/init_node_info.hpp"
+
+namespace InferenceEngine {
+namespace details {
+
+NetworkBatchAbility isNetworkBatchable(const CNNNetwork& orig_network,
+                                       const std::string& deviceNameWithoutBatch,
+                                       bool strictly_track_dims) {
+    CNNNetwork clonedNetwork(cloneNetwork(orig_network));
+    auto function = clonedNetwork.getFunction();
+    // find the batch dim
+    ov::pass::Manager m;
+    m.register_pass<ngraph::pass::InitNodeInfo>();
+    m.register_pass<ov::pass::FindBatch>(true, strictly_track_dims);
+    m.run_passes(function);
+    bool any_batched_inputs = false;
+    // do not reshape/re-batch originally batched networks and when there are no inputs with the N* layouts
+    // input(s) should have the batch dim as the first dim or none (current limitation of the auto-batching impl)
+    const auto& params = function->get_parameters();
+    for (size_t input_id = 0; input_id < params.size(); input_id++) {
+        const auto& input = params[input_id];
+        const auto& shape = input->get_partial_shape();
+        // currently no plugin support batched execution for dynamic networks
+        if (shape.is_dynamic())
+            return NetworkBatchAbility::NO;
+        // check the batch dim: either 0th (and the original batch size of 1) or none
+        if (shape.size() && ov::DimensionTracker::get_label(shape[0])) {
+            const auto& static_shape = input->get_shape();
+            if (static_shape[0] != 1)
+                return NetworkBatchAbility::NO;
+            else
+                any_batched_inputs = true;
+        } else {
+            // if the 0-th dim is not for the batch, then we support only the case when NONE dimension is batch
+            for (size_t s = 1; s < shape.size(); s++)
+                if (ov::DimensionTracker::get_label(shape[s]))
+                    return NetworkBatchAbility::NO;
+        }
+    }
+    if (!any_batched_inputs)
+        return NetworkBatchAbility::NO;
+
+    for (auto&& node : orig_network.getFunction()->get_ops())
+        node->get_rt_info()["affinity"] = "BATCH";  // default affinity (ignored if HETERO is not triggered)
+    // have to execute the DetectionOutput separately (without batching)
+    // as this layer does mix-in the values from the different inputs (batch id)
```

Comment:
@jane-intel Hi, can you help me understand why DO layer does mix-in from different batches?

Reply:
Please see documentation https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/detection/DetectionOutput_8.md

![image](https://user-images.githubusercontent.com/55839243/186112733-c62d8a9a-f4a1-4af8-8f14-1ca3108ec637.png)